modules/vaos/app/services/vaos/middleware/vaos_logging.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module VAOS
+  module Middleware
+    class VaosLogging < Faraday::Middleware
+      def initialize(app)
+        super(app)
+      end
+
+      def call(env)
+        start_time = Time.current
+
+        @app.call(env).on_complete do |response_env|
+          log_tags = {
+            jti: jti(env, response_env),
+            status: response_env.status,
+            duration: Time.current - start_time,
+            # service_name: service_name || 'VAOS Generic', # Need to figure out a clean way to do this with headers
+            url: env.url.to_s
+          }
+
+          if response_env.status.between?(200, 299)
+            log(:info, 'VAOS service call succeeded!', log_tags)
+          else
+            log(:warn, 'VAOS service call failed!', log_tags)
+          end
+        end
+      end
+
+      private
+
+      def log(type, message, tags)
+        Rails.logger.send(type, message, tags)
+      end
+
+      def decode_jwt(token)
+        VAOS::JWT.decode(token, rsa_private.public_key, false, algorithm: 'RS512').first
+      end
+
+      def user_session_request?(env)
+        env.url.to_s.include?('users/v2/session?processRules=true') ? true : false
+      end
+
+      def rsa_private
+        OpenSSL::PKey::RSA.new(File.read(Settings.va_mobile.key_path))
+      end
```

Comment:
Reading a file on disk + initializing an OpenSSL RSA private key seems like an expensive thing to do on every request. Any merit to memoizing this as a class variable instead?

Reply:
so I don't think that code actually gets invoked seeing as we're not verifying the signature. The JWT is not encrypted, it's just signed and to get the JWT we do not currently verify signature.

Reply:
Starting on line 15: `jti()` --> `decode_jwt()` --> `rsa_private.public_key` would execute this code, no?

Reply:
yes, you're right. I've changed the code. And I opened up a ticket to investigate memoizing that as part of user-service as well. Very good catch.